paged/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+//! Welcome to `paged`, a wrapper struct for page alignement and padding!
+//!
+//! When interacting with kernels, it is often required that the references
+//! passed be page-aligned and page-sized. However, specifying this
+//! alignment on the type itself imposes runtime overhead when using these
+//! types in userspace. This can be particularly painful for small types.
+//!
+//! The `paged` crate solves this problem with a convenient wrapper type.
+//! For example:
+//!
+//! ```
+//! use core::mem::*;
+//! use paged::*;
+//!
+//! struct Foo(u32, u32);
+//!
+//! assert_eq!(size_of::<Foo>(), 8);
+//! assert_eq!(align_of::<Foo>(), align_of::<u32>());
+//! assert_eq!(size_of::<Page<Size4k, Foo>>(), 4096);
+//! assert_eq!(align_of::<Page<Size4k, Foo>>(), 4096);
+//!
+//! struct Bar(u32, [u8; 4096]);
+//!
+//! assert_eq!(size_of::<Bar>(), size_of::<(u32, [u8; 4096])>());
+//! assert_eq!(align_of::<Bar>(), align_of::<u32>());
+//! assert_eq!(size_of::<Page<Size4k, Bar>>(), 8192);
+//! assert_eq!(align_of::<Page<Size4k, Bar>>(), 4096);
+//! ```
+//!
+//! This means that you get the alignment and padding when you need it,
```

Comment:
Would you mind pointing out (again) what _this_ refers to? I expected an aha-effect but didn't experience one, so I'm likely misunderstanding something.